scripts/Crawlers/BaseCrawler.py

```diff
@@ -552,3 +558,22 @@ def _check_dats_present(self, directory):
                         return os.path.join(file_path, subfile_name)
             elif file_name.lower() == "dats.json":
                 return file_path
+
+    def _add_source_data_submodule_if_derivedfrom_conp_dataset(
+        self, dats_json, dataset_dir
+    ):
+        f = open(dats_json)
+        metadata = json.loads(f.read())
+        f.close()
+
+        source_dataset_link = None
+        source_dataset_id = None
+        for property in metadata["extraProperties"]:
+            if property["category"] == "derivedFrom":
+                source_dataset_link = property["values"][0]["value"]
+            if property["category"] == "parent_dataset_id":
+                source_dataset_id = property["values"][0]["value"]
+
+        if "github.com" in source_dataset_link:
```

Comment:
I think this will break if `source_dataset_link` is not iterable, e.g. if it is `None`. This might happen if no `property["category"] == "derivedFrom"`. You could make that the first condition, i.e. `if source_dataset_link is not None and ...`.

Reply:
good catch. Done in the next commit